waiter/src/waiter/scheduler/kubernetes.clj

```diff
@@ -264,12 +265,35 @@
                      :revision-timestamp {:pod revision-timestamp :rs rs-revision-timestamp}})
           true))))
 
+;; forward declaration of the hard-delete-service-instance function
+(declare hard-delete-service-instance)
+
+(let [kill-frequently-restarting-pods-thread-pool (Executors/newFixedThreadPool 4)]
+  (defn kill-frequently-restarting-pods
```

Comment:
Since this function only accepts a single target instance (pod), it should be named `kill-frequently-restarting-pod` (no trailing `s`).